### ISSUE TITLE
Video prefetch and store in cache

### ISSUE THREAD

**Issue by slavko-lukic** [TheWidlarzGroup/react-native-video#3318]
# Feature Request

Function that will allow prefetching video similarly like you can prefetch images.
Prefetched video will be stored in cache and it will ensure that there will be no buffering/loading delay later when you need it.

## Why it is needed

Depends a lot on app use case, but in some apps you maybe don't want to stop the flow while video downloads.

### Code sample

I imagine it would work something like this.

```javascript
Video.prefetch("someUrl").then((isSuccess) => {
    // do something
})

// or

const isSuccess = await Video.prefetch("someUrl");
```

Param could be string or array of strings if you want to prefetch more than one url.

**Comment by freeboub** [comment]
Good idea, but more props are needed. I think similar props than the source would be better. If possible should be something similar to the download to go feature

**Comment by wydv21** [comment]
@slavko-lukic This helps for now https://github.com/zhigang1992/react-native-video-cache#readme

**Comment by yash-vaghela21** [comment]
@slavko-lukic This helps for now - https://github.com/zhigang1992/react-native-video-cache#readme
not prefetching but cache the video once it played.

**Comment by derynusaputra** [comment]
> @slavko-lukic This helps for now - https://github.com/zhigang1992/react-native-video-cache#readme not prefetching but cache the video once it played.

I have tried this method, but it doesn't work, can you help me? I'm making a TikTok-like project using hls video files. The problem is that when you swipe down, first start the video loading first.

**Comment by gideon-uiby** [comment]
Any update on the feature request?

**Comment by hari-mohan-choudhary** [comment]
@gideon-uiby did you able to find any solution. I was looking same solution

**Comment by freeboub** [comment]
No progress on this point, it is an heavy task to implement (especially for dash and m3u). maybe you can join TheWidlzarGroup to discuss about this subject !
